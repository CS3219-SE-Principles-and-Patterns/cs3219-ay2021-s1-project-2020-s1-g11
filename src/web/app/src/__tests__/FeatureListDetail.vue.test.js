import {createLocalVue, shallowMount} from '@vue/test-utils'
import ElementUI from 'element-ui';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import FeatureListDetail from "./../components/homePageDetail/FeatureListDetail.vue";

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);
localVue.use(ElementUI);

describe("EmptySection.vue", () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        isPageLoading: false,
        userInfo: {
          isApiError: false,
          isLogin: true,
        },
        dbMetaData: {
          entitiesStatus: {
            isLoading: false,
          }
        }
      }
    })
  });

  it('renders correctly', () => {
    const wrapper = shallowMount(FeatureListDetail, {store, localVue})
    expect(wrapper.element).toMatchSnapshot()
  });
});