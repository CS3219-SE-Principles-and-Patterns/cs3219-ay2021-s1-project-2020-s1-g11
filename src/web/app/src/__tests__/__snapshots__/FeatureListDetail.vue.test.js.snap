// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptySection.vue renders correctly 1`] = `
<el-main-stub>
  <div
    class="page"
  >
    <!---->
     
    <div
      class="options-list"
    >
      <h1>
         Exploring Chairvise 
      </h1>
       
      <el-row-stub
        align="middle"
        gutter="16"
        justify="center"
        tag="div"
        type="flex"
      >
        <zoom-x-transition-stub
          delay="600"
          duration="500"
          origin=""
          styles="[object Object]"
          tag="span"
        >
          <el-col-stub
            lg="8"
            md="8"
            sm="24"
            span="24"
            style="display: none;"
            tag="div"
            xl="8"
          >
            <el-card-stub
              class="feature-card"
              shadow="hover"
            >
              <img
                src="@/assets/upload.png"
              />
               
              <el-button-stub
                class="button"
                icon=""
                nativetype="button"
                type="primary"
              >
                
                Import Data
              
              </el-button-stub>
            </el-card-stub>
          </el-col-stub>
        </zoom-x-transition-stub>
         
        <zoom-x-transition-stub
          delay="600"
          duration="500"
          origin=""
          styles="[object Object]"
          tag="span"
        >
          <el-col-stub
            lg="8"
            md="8"
            sm="24"
            span="24"
            style="display: none;"
            tag="div"
            xl="8"
          >
            <el-card-stub
              class="feature-card"
              shadow="hover"
            >
              <img
                src="@/assets/stadistics.png"
              />
               
              <el-button-stub
                class="button"
                icon=""
                nativetype="button"
                type="primary"
              >
                
                Analyse Data 
              
              </el-button-stub>
            </el-card-stub>
          </el-col-stub>
        </zoom-x-transition-stub>
         
        <zoom-x-transition-stub
          delay="600"
          duration="500"
          origin=""
          styles="[object Object]"
          tag="span"
        >
          <el-col-stub
            lg="8"
            md="8"
            sm="24"
            span="24"
            style="display: none;"
            tag="div"
            xl="8"
          >
            <el-card-stub
              class="feature-card"
              shadow="hover"
            >
              <img
                src="@/assets/schedule.png"
              />
               
              <el-button-stub
                class="button"
                icon=""
                nativetype="button"
                type="primary"
              >
                
                Track Conference
              
              </el-button-stub>
            </el-card-stub>
          </el-col-stub>
        </zoom-x-transition-stub>
      </el-row-stub>
    </div>
  </div>
</el-main-stub>
`;
